great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -724,7 +724,8 @@ def delete_datasource(
             raise ValueError(f"Datasource {datasource_name} not found")
 
         if save_changes:
-            self._datasource_store.delete_by_name(datasource_name)  # type: ignore[attr-defined]
+            datasource_config = datasourceConfigSchema.load(datasource.config)
+            self._datasource_store.delete(datasource_config)  # type: ignore[attr-defined]
```

Comment:
Can this `attr-defined` error be easily resolved?

Reply:
I don't believe so - the `self._datasource_store` attr is only ever instantiated by child classes. We likely want to change this but it may take some more work